gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -346,4 +346,15 @@ def self.inherited(child_class)
 
     raise "Inheriting from children of T::Enum is prohibited" if self != T::Enum
   end
+
+  # Marshal support
+  sig {params(_level: Integer).returns(String)}
+  def _dump(_level)
```

Comment:
Let's not go as low-level.
Can we do `marshal_dump` instead?

ditto about `marshal_load` instead of `_load`

Reply:
My impression is that the difference of (`marshal_dump` and `marshal_load`) vs(`_dump` and `_load`) is who's doing the object allocation, `Marshal` or our code respectively.

In our case, I think `_load` and `_dump` are more appropriate since what we want to do is reuse the already existing allocated object, given that object identity comparison is how we implement the case equality operator (`===`)

Reply:
yeah i think we need (and want) `_dump` here